roles/platform/tasks/setup_gcp_authz.yml

```diff
@@ -14,161 +14,166 @@
+          condition: "expression=resource.name == '//storage.googleapis.com/projects/_/buckets/{{ plat__gcp_storage_location_data }}',title='{{ plat__gcp_ranger_audit_identity_name }}-fulladmin'"
+        # ID Broker  / Assumer Role
+        - member: "serviceAccount:{{ plat__gcp_idbroker_identity_name }}@{{ plat__gcp_project }}.iam.gserviceaccount.com"
+          role: "{{ plat__gcp_roles.iam_workload_identity_user }}"
+          condition: "expression=resource.name.startsWith('{{ plat__namespace }}'),title='{{ plat__gcp_idbroker_identity_name }}'"
+        - member: "serviceAccount:{{ plat__gcp_idbroker_identity_name }}@{{ plat__gcp_project }}.iam.gserviceaccount.com"
+          role: "{{ plat__gcp_roles.iam_service_account_user }}"
+          condition: "None"
+        - member: "serviceAccount:{{ plat__gcp_idbroker_identity_name }}@{{ plat__gcp_project }}.iam.gserviceaccount.com"
+          role: "{{ plat__gcp_roles.iam_service_account_token_creator }}"
+          condition: "None"
+      command: >
+        gcloud projects
+        add-iam-policy-binding {{ plat__gcp_project }}
+        --member={{ __gcp_binding_item.member |quote }}
+        --role={{ __gcp_binding_item.role |quote }}
+        --condition={{ __gcp_binding_item.condition |quote }}
 
-- name: Add Service Accounts to Storage Policies for Buckets
-  loop_control:
-    loop_var: __gcp_pol_item
-  loop:
-    - account: "serviceAccount:{{ plat__gcp_log_identity_name }}@{{ plat__gcp_project }}.iam.gserviceaccount.com:admin"
-      bucket: "{{ plat__gcp_storage_location_logs }}"
-    - account: "serviceAccount:{{ plat__gcp_datalakeadmin_identity_name }}@{{ plat__gcp_project }}.iam.gserviceaccount.com:admin"
-      bucket: "{{ plat__gcp_storage_location_data }}"
-    - account: "serviceAccount:{{ plat__gcp_ranger_audit_identity_name }}@{{ plat__gcp_project }}.iam.gserviceaccount.com:admin"
-      bucket: "{{ plat__gcp_storage_location_data }}"
-  command: >
-    gsutil iam
-    ch {{ __gcp_pol_item.account |quote }}
-    gs://{{ __gcp_pol_item.bucket |quote }}
\ No newline at end of file
+    - name: Add Service Accounts to Storage Policies for Buckets
+      loop_control:
+        loop_var: __gcp_pol_item
+      loop:
+        - account: "serviceAccount:{{ plat__gcp_log_identity_name }}@{{ plat__gcp_project }}.iam.gserviceaccount.com:admin"
+          bucket: "{{ plat__gcp_storage_location_logs }}"
+        - account: "serviceAccount:{{ plat__gcp_datalakeadmin_identity_name }}@{{ plat__gcp_project }}.iam.gserviceaccount.com:admin"
+          bucket: "{{ plat__gcp_storage_location_data }}"
+        - account: "serviceAccount:{{ plat__gcp_ranger_audit_identity_name }}@{{ plat__gcp_project }}.iam.gserviceaccount.com:admin"
+          bucket: "{{ plat__gcp_storage_location_data }}"
+      command: >
+        gsutil iam
+        ch {{ __gcp_pol_item.account |quote }}
+        gs://{{ __gcp_pol_item.bucket |quote }}
+
+- pause:
```

Comment:
Need to remove the trailing `pause` here - presume it's leftover from debugging.

Reply:
bah.

Reply:
Fixed. Please can you review @jimright